File: feature/lego_list/src/main/java/com/example/lego/feature/legolist/LegoListViewModel.kt
package com.example.lego.feature.legolist

import androidx.lifecycle.ViewModel
import dagger.hilt.android.lifecycle.HiltViewModel
import javax.inject.Inject

import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow

data class LegoUniverse(
    val id: String,
    val title: String,
    val isAvailable: Boolean
)

@HiltViewModel
class LegoListViewModel @Inject constructor() : ViewModel() {
    private val _universes = MutableStateFlow<List<LegoUniverse>>(emptyList())
    val universes: StateFlow<List<LegoUniverse>> = _universes

    init {
        _universes.value = listOf(
            LegoUniverse("harry_potter", "Harry Potter Universe", true),
            LegoUniverse("star_war", "Star War Universe", false),
            LegoUniverse("star_trek", "Star Trek Universe", false),
            LegoUniverse("dragon_ball", "Dragon Ball-Z Universe", false)
        )
    }
}

File: Sources/Features/LegoList/LegoListViewModel.swift
import SwiftUI

struct LegoUniverse: Identifiable {
    let id: String
    let name: String
    let isAvailable: Bool
}

class LegoListViewModel: ObservableObject {
    @Published var universes: [LegoUniverse] = []
    
    init() {
        universes = [
            LegoUniverse(id: "harry_potter", name: "Harry Potter Universe", isAvailable: true),
            LegoUniverse(id: "star_wars", name: "Star Wars Universe", isAvailable: true),
            LegoUniverse(id: "star_trek", name: "Star Trek Universe", isAvailable: true),
            LegoUniverse(id: "dragon_ball", name: "Dragon Ball-Z Universe", isAvailable: true)
        ]
    }
}
